spunk2argus

sbob909 opened the pull request:
New example, allows for more flexible Splunk queries and assignment of
result set fields to Argus scope, metrics, and tags

hdara-sfdc commented on examples/splunk2argus.py:
```diff
@@ -0,0 +1,234 @@
+
+        if not index:
+            logging.warn("Skipping row without index: %s", row)
+            continue
```
Lines 170 to 174 seem to be leftover code? Rather, I guess you would want a generalized check inside the loop at line 181.

hdara-sfdc commented on examples/splunk2argus.py:
```diff
@@ -0,0 +1,234 @@
+        # create final scope, substitute keys into the scope
+        rowScope = opts.argusscope
+        for keyName in keyNames:
+            rowScope = rowScope.replace("{"+keyName+"}",row[keyName])
```
Check for existence of `keyName` in `row`.

hdara-sfdc commented on examples/splunk2argus.py:
```diff
@@ -0,0 +1,234 @@
+        # create tags
+        tag_dict = {}
+        for tagName in tagNames:
+            tag_dict[tagName] = row[tagName]
```
Check for existence of `tagName` in `row`.

hdara-sfdc commented on examples/splunk2argus.py:
```diff
@@ -0,0 +1,234 @@
+
+            m_key = (col)
+
+            val = row[col]
```
Check for existence of `col` in `row`.

hdara-sfdc commented on examples/splunk2argus.py:
```diff
@@ -0,0 +1,234 @@
+            if "." in val:
+                val = float(val)
+            else:
+                val = int(val)
```
If someone misconfigures tag columns as metric columns, it is possible to see non-numeric values here, so if we don't check for `val.isdigit()`, the end user would see an abrupt unhelpful exception.

sbob909 commented:
@hdara-sfdc thanks for the awesome feedback! I think I addressed all of your suggestions (and more) in the latest commit to my branch. Please review again (remember, I'm a python noob) and merge if there aren't any other findings.
